server/channels/api4/preference.go

```diff
@@ -128,6 +129,14 @@ func updatePreferences(c *Context, w http.ResponseWriter, r *http.Request) {
 			}
 		}
 
+		if pref.Category == model.PreferenceCategorySidebarSettings && pref.Name == model.PreferenceLimitVisibleDmsGms {
```

Comment:
Could this be done in `Preference.IsValid`?

Reply:
Moved to `Preference.IsValid`